### ISSUE TITLE
Eager initialization is not sufficiently eager.

### ISSUE THREAD

**Issue by e-kolpakov** [akka/akka-persistence-cassandra#350]
Steps to reproduce: 

1. Enable debug logging `akka.loglevel = DEBUG`
2. Filter log records by "Create plugin" (i.e. `tail -f application.log | grep "Create plugin"`)
3. Configure plugin to eagerly initialize journal and snapshot store:

```
akka.persistence {
    journal.auto-start-journals = [ ${akka.persistence.journal.plugin} ]
    snapshot-store.auto-start-snapshot-stores = [ ${akka.persistence.snapshot-store.plugin} ]
}
```

4. Start the application.
5. Trigger actor recovery.
6. Observe the follwoing:
6.1. `cassandra-journal` and `cassandra-snapshot-store` are eagerly initialized at application startup
6.2. `cassandra-query-journal` is lazily initialized only after first actor triggers recovery.
6.3. Example logs:

```
18-05-14T14:37:52.814 DEBUG spatcher-3 a.p.Persistence(akka://actor-system) - Create plugin: cassandra-journal akka.persistence.cassandra.journal.CassandraJournal
18-05-14T14:37:52.875 DEBUG patcher-22 a.p.Persistence(akka://actor-system) - Create plugin: cassandra-snapshot-store akka.persistence.cassandra.snapshot.CassandraSnapshotStore
**18-05-14T14:38:37.634** DEBUG spatcher-4 a.p.q.PersistenceQuery - Create plugin: cassandra-query-journal akka.persistence.cassandra.query.CassandraReadJournalProvider
```

**Real world impact:** since `CassandraReadJournalProvider` create another batch of connections to Cassandra, recovery of the first actor is delayed for a few seconds. This is especially sensible in combination with Cluster Sharding and under low latency and high throughput requirements, as persistent actors handed off to a newly joined instance become unavailable for a few seconds.

**Comment by patriknw** [comment]
@e-kolpakov Thanks for reporting. I think this is specific for akka-persistence-cassandra. Can you create the issue [there](https://github.com/akka/akka-persistence-cassandra/issues/new) instead?

**Comment by johanandren** [comment]
@patriknw this is the akka/akka-persistence-cassandra repo?

**Comment by patriknw** [comment]
Doh, sorry.

**Comment by chbatey** [comment]
Note that we will make it the same connection for 1.0 when we break the config https://github.com/akka/akka-persistence-cassandra/issues/81

**Comment by chbatey** [comment]
Closed via #351

**Comment by e-kolpakov** [comment]
Thanks guys!
